collect_app/src/main/java/org/odk/collect/android/preferences/screens/BasePreferencesFragment.kt

```diff
@@ -43,9 +44,12 @@ abstract class BasePreferencesFragment : PreferenceFragmentCompat(), OnSettingCh
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        if (activity is CollectAbstractActivity) {
-            (activity as CollectAbstractActivity).initToolbar(preferenceScreen.title)
+        requireActivity().let {
+            if (it is AppCompatActivity) {
```

Comment:
Do we need this check? I think we use `AppCompatActivity` pretty much everywhere now.

Reply:
We'd end up with a **technically** unsafe cast in the code without it as `requireActivity()` returns `FragmentActivity` rather than `AppCompatActivity` but as you say there's probably nowhere that we'd actually see a problem. I don't have strong feelings either way when it comes to the safe code vs exploding if we're using a non `AppCompatActivity`. What do you think?

Reply:
I don't have strong feelings but I would like to avoid checks like that. It seems now like it's just one place but it can be anywhere too. Maybe that new `AppBarUtils` should take `Activity` and there check if it's `AppCompatActivity`?

Reply:
> Maybe that new AppBarUtils should take Activity and there check if it's AppCompatActivity

I like that! Keeps the code safe but also means we don't need to check everywhere. I'll make that change.